analyzer_test.go

```diff
@@ -70,8 +71,38 @@ func TestAnalyzer(t *testing.T) {
 			pkg:                  "f",
 			disallowExtraAliases: true,
 		},
+		{
+			desc: "regexp with non capturing groups",
+			pkg:  "g",
+			aliases: stringMap{
+				"knative.dev/serving/pkg/(?:apis/)?(\\w+)(?:/v[\\w\\d]+)?": "k$1",
+			},
+		},
+	}
+
+	makeAnalyzer := func() *analysis.Analyzer {
+		cnf := Config{
+			RequiredAlias: make(map[string]string),
+		}
+		return &analysis.Analyzer{
+			Flags: flags(&cnf),
+			Run: func(pass *analysis.Pass) (interface{}, error) {
+				return runWithConfig(&cnf, pass)
+			},
+			Requires: []*analysis.Analyzer{inspect.Analyzer},
+		}
 	}
 
+	t.Run("wrong usage", func(t *testing.T) {
```

Comment:
I'd either move this test to its own top-level function (e.g. `TestIncorrectFlags`), or add it to the existing table in this test (e.g. by adding an `expectAliasFlagErr` field to the rows). I'd also personally be fine if we exported the `stringMap` type and relied on a unit test (without constructing an analyser) for that to test this case.